packages/vaadin-combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -550,16 +550,32 @@ export const ComboBoxMixin = (subclass) =>
      */
     _onEscape(e) {
```

Comment:
`_onEscape` logic is unnecessarily complex to begin with. Something like

```js
if (this.opened || this.value !== this._inputElementValue) {
  this._focusedIndex = -1;
  this.cancel();
} else if (this.clearButtonVisible) {
  this._clear();
}
```

would be nicer and in line with https://github.com/vaadin/web-components/issues/1332 but would also change the current UX and thus require changes to existing tests. Would prefer doing breaking changes separately and have this be merely a bug fix.

Reply:
Good point 💯